sources/packages/api/src/services/bceid/bceid.service.ts

```diff
@@ -0,0 +1,152 @@
+import { Injectable } from "@nestjs/common";
+import { BasicAuthSecurity, Client, createClientAsync } from "soap";
+import { InjectLogger } from "../../common";
+import { BCeIDConfig } from "../../types/config";
+import { LoggerService } from "../../logger/logger.service";
+import { ConfigService } from "../config/config.service";
+import { AccountDetails } from "./account-details.model";
+import {
+  BCeIDAccountTypeCodes,
+  ResponseBase,
+  ResponseCodes,
+} from "./bceid.models";
+
+/**
+ * Manage the execution of SOAP requests to the BCeID Web Service as per described
+ * on https://sminfo.gov.bc.ca/ based on the documentation available on
+ * "BCeID Client - New Web Services - Developers Guide - V2.10.0 (for BCeID WS V10).docx"
+ */
+@Injectable()
+export class BCeIDService {
+  private bceidConfig: BCeIDConfig;
+  constructor(private readonly config: ConfigService) {
+    this.bceidConfig = this.config.getConfig().bceid;
+  }
+
+  /**
+   * This method allows you to retrieve the details of a single BCeID account.
+   * This method respects the information sharing rules. To perform a query, there must be an
+   * information sharing rule for the online service (onlineServiceId) with the specified requester
+   * user account type (requesterAccountTypeCode) and target user account type (accountTypeCode).
+   * The information returned by a query is restricted to the properties included in the information sharing rules.
+   * @param userName User name associated with the account whose details are being queried.
+   * @returns account details if the account was found, otherwise null.
+   */
+  public async getAccountDetails(userName: string): Promise<AccountDetails> {
+    var client = await this.getSoapClient();
+    // SOAP call body to execute the getAccountDetail request.
+    var body = {
+      accountDetailRequest: {
+        onlineServiceId: this.bceidConfig.onlineServiceId,
+        // Internal indicate that the user being provided
+        // on requesterUserGuid is an user that belongs
+        // to the internal gov network.
+        requesterAccountTypeCode: BCeIDAccountTypeCodes.Internal,
+        // The user guid of the user on the internal gov network.
+        requesterUserGuid: this.bceidConfig.requesterUserGuid,
+        userId: userName,
+        // Type of the user account to search for the userId
+        // parameter provided above. Only business BCeID are
+        // on the scope of the application.
+        accountTypeCode: BCeIDAccountTypeCodes.Business,
```

Comment:
Am confused between accountTypeCode and Request AcountTypeCode. Lets talk on this

Reply:
I believe that you are talking about requesterAccountTypeCode and accountTypeCode, right?
- requesterAccountTypeCode: identification from the user doing the request. It will be used by BCeID Web Service to check permissions to access data.
- accountTypeCode: the type of the user account that we provided in the parameter userId.